presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -446,6 +446,18 @@ public void close()
         orcDataSink.close();
     }
 
+    private void resetInternalWriters()
+    {
+        columnWriters.forEach(ColumnWriter::reset);
+        closed = false;
```

Comment:
It might keep the original logic that once `OrcWriter` is closed, it will never be opened.